developer/best_practices.md

```diff
@@ -0,0 +1,188 @@
+very mature software available that is designed for this very use case.
+
+Therefore, adopters of EDC have to general options to consider:
+
+1. develop a custom `AuthenticationService`, that integrates with an IDP
+2. use a dedicated API gateway (recommended)
+
+Both these options are viable, and may have merit depending on the use case.
+
+### 1.5 Docker builds
+
+As Docker is a very popular method to build and ship applications, we put forward the following recommendations:
+
+- use official Eclipse Temurin base images for Java
+- use dedicated non-root users: in your Dockerfile, add the following lines
+   ```dockerfile
+    ARG APP_USER=docker  
+    ARG APP_UID=10100  
+    RUN addgroup --system "$APP_USER"
+    RUN adduser \  
+	     --shell /sbin/nologin \  
+	     --disabled-password \  
+	     --gecos "" \  
+	     --ingroup "$APP_USER" \  
+	     --no-create-home \  
+	     --uid "$APP_UID" \  
+	     "$APP_USER"
+    
+    USER "$APP_USER"
+   ```
+
+### 1.6 Use proper database security
+
+Database connections are secured with a username and a password. Please choose non-default users and strong passwords.
+In addition, database credentials should be stored in an HSM (vault).
+
+Further, the roles of the technical user for the connector should be limited to `SELECT`, `INSERT`, `UPDATE`,
+and `DELETE`. There is no reason for that user to have permissions to modify databases, tables, permissions or execute
+other DDL statements.
+
+### 1.7 Store sensitive data in a Vault
+
+While the default behaviour of EDC is that configuration values are taken either from environment variables, system
+properties or from configuration extensions, it is highly recommended to store sensitive data in a `Vault` when
+developing EDC extensions.
+
+Here is a (non-exhaustive) list of examples of such sensitive values:
+
+- database credentials
+- cryptographic keys, e.g. private keys in an asymmetric key pair
+- symmetric keys
+- API keys/tokens
+- credentials for other third-party services, even if temporary
+
+Sensitive values should not be passed through multiple layers of code. Instead, they should be referenced by their
+alias, and be resolved from the `Vault` wherever they are used. Do not store sensitive data as class members but use
+local variables that are garbage-collected when leaving execution scope.
+
+## 2. General recommendations
+
+### 2.1 Use only official releases
+
+We recommend using _only official releases_ of our components. The latest version can be obtained from the
+project's [GitHub releases page](https://github.com/eclipse-edc/Connector/releases) and the modules are available from
+MavenCentral.
+
+Snapshots are less stable, less tested and less reliable than release versions and the make for non-repeatable builds.
+
+That said, we realize that sometimes living on the bleeding edge of technology is thrilling, or in some circumstances
```

Comment:
We could reference https://eclipse-edc.github.io/docs/#/documentation/developer/releases here.